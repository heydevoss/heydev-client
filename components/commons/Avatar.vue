<template>
  <img :src="avatarUrl || require('~/assets/avatar/default.png')" />
</template>

<script>
export default {
  name: 'Avatar',
  props: {
    avatarUrl: String
  }
}
</script>
<style lang="scss" scoped>
img {
  border-radius: 100%;
  height: 300px;
  object-fit: cover;
  width: 300px;
}
</style>
