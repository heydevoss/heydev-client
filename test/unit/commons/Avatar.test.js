import { shallowMount } from '@vue/test-utils'
import Avatar from '@/components/commons/Avatar.vue'

describe('LoginButton', () => {
  const wrapper = shallowMount(Avatar)

  it('should be a Vue instance and render correctly', () => {
    expect(wrapper.isVueInstance()).toBeTruthy()
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should contain <img/> with default', () => {
    expect(wrapper.html()).toContain('<img src=""')
  })
  it('should contain <img/> with src passed', () => {
    wrapper.setProps({
      avatarUrl: '@/assets/avatar/default.png'
    })
    expect(wrapper.html()).toContain('<img src="@/assets/avatar/default.png"')
  })
})
