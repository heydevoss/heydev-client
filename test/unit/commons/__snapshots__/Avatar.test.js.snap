// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Avatar/> should be a Vue instance and render correctly 1`] = `
<img
  src=""
/>
`;
